lib/vanagon/platform/windows.rb

```diff
@@ -107,21 +107,24 @@ def generate_msi_package(project)
         target_dir = project.repo ? output_dir(project.repo) : output_dir
         cg_name = "ProductComponentGroup"
         dir_ref = "INSTALLDIR"
+        candle_flags =  "-dPlatform=#{@architecture} -arch #{@architecture} -ext WiXUtilExtension -ext WixUIExtension"
         # Actual array of commands to be written to the Makefile
         ["mkdir -p output/#{target_dir}",
-        "mkdir -p $(tempdir)/{staging,wix}",
-        "#{@copy} -r wix/* $(tempdir)/wix/",
-        "gunzip -c #{project.name}-#{project.version}.tar.gz | '#{@tar}' -C '$(tempdir)/staging' --strip-components 1 -xf -",
-        # Run the Heat command in a single pass
-        # Heat command documentation at: http://wixtoolset.org/documentation/manual/v3/overview/heat.html
-        #   dir <directory> - Traverse directory to find all sub-files and directories.
-        #   -ke             - Keep Empty directories
-        #   -cg             - Component Group Name
-        #   -gg             - Generate GUIDS now
-        #   -dr             - Directory reference to root directories (cannot contains spaces e.g. -dr MyAppDirRef)
-        #   -sreg           - Suppress registry harvesting.
-        #   -var <variable> - Substitute File/@Source="SourceDir" with a preprocessor or a wix variable
-        "cd $(tempdir); \"$$WIX/bin/heat.exe\" dir staging -v -ke -indent 2 -cg #{cg_name} -gg -dr #{dir_ref} -t wix/#{project.name}.filter.xslt -sreg -var var.StageDir -out wix/#{project.name}-harvest.wxs",
+          "mkdir -p $(tempdir)/{staging,wix/wixobj}",
+          "#{@copy} -r wix/* $(tempdir)/wix/",
+          "gunzip -c #{project.name}-#{project.version}.tar.gz | '#{@tar}' -C '$(tempdir)/staging' --strip-components 1 -xf -",
+          # Run the Heat command in a single pass
+          # Heat command documentation at: http://wixtoolset.org/documentation/manual/v3/overview/heat.html
+          #   dir <directory> - Traverse directory to find all sub-files and directories.
+          #   -ke             - Keep Empty directories
+          #   -cg             - Component Group Name
+          #   -gg             - Generate GUIDS now
+          #   -dr             - Directory reference to root directories (cannot contains spaces e.g. -dr MyAppDirRef)
+          #   -sreg           - Suppress registry harvesting.
+          "cd $(tempdir); \"$$WIX/bin/heat.exe\" dir staging -v -ke -indent 2 -cg #{cg_name} -gg -dr #{dir_ref} -t wix/#{project.name}.filter.xslt -sreg -out wix/#{project.name}-harvest.wxs",
+          # Apply Candle command to all *.wxs files - generates .wixobj files in wix directory.
+          # cygpath conversion is necessary as candle is unable to handle posix path specs
+          "cd $(tempdir)/wix/wixobj; for wix_file in `find $(tempdir)/wix -name \'*.wxs\'`; do \"$$WIX/bin/candle.exe\" #{candle_flags} $$(cygpath -aw $$wix_file) ; done",
```

Comment:
@melissa Have incorporated your suggestions here.
Had to use the $(cygwin -aw) command to fixup the file path for Candle.